source/Module/STIG/Functions.Checklist.ps1

```diff
@@ -166,219 +255,227 @@ function New-StigCheckList
-            $manualCheck = $manualCheckData | Where-Object {$_.VulID -eq $VID}
+                $writer.WriteStartElement("VULN_ATTRIBUTE")
+                $writer.WriteString($attribute.Name)
+                $writer.WriteEndElement(<#VULN_ATTRIBUTE#>)
 
-            if ($setting)
-            {
-                $status = $statusMap['NotAFinding']
-                $comments = "To be addressed by PowerStig MOF via $setting"
-                $findingDetails = Get-FindingDetails -Setting $setting
+                $writer.WriteStartElement("ATTRIBUTE_DATA")
+                $writer.WriteString($attribute.Value)
+                $writer.WriteEndElement(<#ATTRIBUTE_DATA#>)
 
+                $writer.WriteEndElement(<#STIG_DATA#>)
             }
-            elseif ($manualCheck)
-            {
-                $status = $statusMap["$($manualCheck.Status)"]
-                $findingDetails = $manualCheck.Details
-                $comments = $manualCheck.Comments
-            }
-            else
-            {
-                $status = $statusMap['NotReviewed']
-            }
-        }
-        elseif ($PSCmdlet.ParameterSetName -eq 'result')
-        {
-            $manualCheck = $manualCheckData | Where-Object -FilterScript {$_.VulID -eq $VID}
-            # If we have manual check data, we don't need to look at the configuration
-            if ($manualCheck)
+
+            if ($PSCmdlet.ParameterSetName -eq 'mof')
             {
-                $status = $statusMap["$($manualCheck.Status)"]
-                $findingDetails = $manualCheck.Details
-                $comments = $manualCheck.Comments
+                $setting = Get-SettingsFromMof -ReferenceConfiguration $ReferenceConfiguration -Id $vid
+                $manualCheck = $manualCheckData.stigManualChecklistData.stigRuleData | Where-Object {$_.STIG -eq $stigFileName -and $_.ID -eq $vid}
+
+                if ($setting)
+                {
+                    $status = $statusMap['Open']
+                    $comments = "To be addressed by PowerStig MOF via $setting"
+                    $findingDetails = Get-FindingDetails -Setting $setting
+
+                }
+                elseif ($manualCheck)
+                {
+                    $status = $statusMap["$($manualCheck.Status)"]
+                    $findingDetails = $manualCheck.Details
+                    $comments = $manualCheck.Comments
+                }
+                else
+                {
+                    $status = $statusMap['NotReviewed']
+                }
             }
-            else
+            elseif ($PSCmdlet.ParameterSetName -eq 'dsc')
             {
-                $setting = Get-SettingsFromResult -DscResult $dscResult -Id $vid
-                if ($setting)
+                $manualCheck = $manualCheckData.stigManualChecklistData.stigRuleData | Where-Object {$_.STIG -eq $stigFileName -and $_.ID -eq $vid}
```

Comment:
-FilterScript for Where-Object is needed

Reply:
Done.